ceph/rados/bluestoretool_workflows.py

```diff
@@ -0,0 +1,301 @@
+    ):
+        """
+        Runs ceph-bluestore-tool commands within OSD container
+        Args:
+            cmd: command that needs to be run
+            osd_id: daemon ID of target OSD
+            format (optional): Output format where ever applicable
+            timeout: Maximum time allowed for execution.
+        Returns:
+            output of respective ceph-bluestore-tool command
+        """
+        osd_node = self.rados_obj.fetch_host_node(
+            daemon_type="osd", daemon_id=str(osd_id)
+        )
+        self.rados_obj.change_osd_state(action="stop", target=osd_id)
+        base_cmd = f"cephadm shell --name osd.{osd_id} --"
+        _cmd = f"{base_cmd} {cmd} --path /var/lib/ceph/osd/ceph-{osd_id}"
+        if format:
+            _cmd = f"{_cmd} --format {format}"
+        try:
+            out, err = osd_node.exec_command(sudo=True, cmd=_cmd, timeout=timeout)
+        except Exception as er:
+            log.error(f"Exception hit while command execution. {er}")
+            raise
+        finally:
+            self.rados_obj.change_osd_state(action="start", target=osd_id)
+        return out
+
+    def help(self, osd_id: int):
+        """Module to run help command with ceph-bluestore-tool to display usage
+         Args:
+            osd_id: OSD ID for which cbt will be executed
+
+        Returns:
+            Output of ceph-bluestore-tool usage
+        """
+        _cmd = "ceph-bluestore-tool --help"
+        return self.run_cbt_command(cmd=_cmd, osd_id=osd_id)
+
+    def run_consistency_check(self, osd_id: int, deep: bool = False):
+        """Module to run consistency check on BlueStore metadata.
+        If deep is specified, also read all object data and verify checksums.
+        Args:
+            osd_id: OSD ID for which cbt will be executed
+        Returns:
+            Returns the output of cbt quick fsck
+        """
+        # Extracting the ceush map from the cluster
+        _cmd = "ceph-bluestore-tool fsck "
+        if deep:
+            _cmd = f"{_cmd} --deep true"
+        return self.run_cbt_command(cmd=_cmd, osd_id=osd_id)
+
+    def run_quick_consistency_check(self, osd_id: int):
+        """Module to run consistency check on BlueStore metadata comparing allocator
+         data (from RocksDB CFB when exists and if not uses allocation-file) with ONodes state.
+        Args:
+            osd_id: OSD ID for which cbt will be executed
+        Returns:
+            Returns the output of cbt quick fsck
+        """
+        _cmd = "ceph-bluestore-tool qfsck"
+        return self.run_cbt_command(cmd=_cmd, osd_id=osd_id)
+
+    def repair(self, osd_id: int):
+        """Module to run a consistency check and repair any errors it can.
+        Args:
+            osd_id: OSD ID for which cbt will be executed
+        Returns:
+            Returns the output of cbt repair cmd
+        """
+        _cmd = "ceph-bluestore-tool repair"
```

Comment:
Is consistency check execution needs to be added here as per docstrings? or if the module is for repair only many it can be reworded.

Reply:
This is a per upstream documentation description of the command - https://docs.ceph.com/en/latest/man/8/ceph-bluestore-tool/#commands

Reply:
Noted